src/Email.elm

```diff
@@ -68,19 +68,29 @@ checkLocal str =
                 || (c == '`')
                 || (c == '{')
                 || (c == '}')
+                || (c == '.')
     in
     if String.isEmpty str then
-        problem "local part of email is empty"
+        problem "local part is empty"
+
+    else if String.startsWith "." str then
+        problem "local part cannot start with ."
+
+    else if String.endsWith "." str then
+        problem "local part cannot end with ."
+
+    else if String.contains ".." str then
+        problem "local part cannot contain .."
```

Comment:
The specific error messages are really nice to end users so they can see why it failed.